instrumentation/httpclient/src/main/java/brave/httpclient/TracingProtocolExec.java

```diff
@@ -81,10 +81,12 @@
     }
 
     @Override public String path() {
+      if (delegate instanceof HttpRequestWrapper) {
```

Comment:
Instead of special casing how we get this value, could we instead unwrap using `getOriginal()` on the wrapper class then follow the flow we already have. I see the benefit here given that URI already will contain the parsed path. Just thinking out loud about maintaining parity

Reply:
Thanks for the idea, I like it, but unfortunately it seems that `getOriginal().getRequestLine()` still contains the host, so we'd need some special casing or more sophisticated URL parsing logic anyways. Best probably to take advantage of the parsing that already happened.